### PR TITLE
Add minimal working notebook showing UNet training 

### PR DESCRIPTION
**Pull request by metazool**
This adds the minimal revisions needed to reproduce Zayad's experiment, mostly in this notebook:

https://github.com/BritishGeologicalSurvey/CoreScore/blob/train-unet/notebooks/Unet-Fastai.ipynb

Next step is to move this out into a package so we can run the experiment using MLflow and I would value any feedback on / input to it at this stage. Particularly about the parameter optimisation - from what I see there is some attempt at tuning hyperparameters but then the notebook trails off into adjusting e.g. learning rate by hand and then visually inspecting the outcome. Any advice on what to try tuning when training with MLFlow and how to record it would be appreciated

Bear in mind this is running on a really small set of training data - 8 labelled images (before validation split) which have been pre-cropped to the core box. I think it would be better to annotate the full images, if we can train on them directly all the better, if we have to crop for better results then we can do that programmatically hopefully with aid from https://github.com/rgmyr/corebreakout

To this end it would be good to use the same set of training images of BGS core as `corebreakout` use for our extra annotations of fragments. https://github.com/rgmyr/corebreakout/releases/tag/v0.2 - the other thing i've been wondering is should we split the label for Rock_Fragment into two, one for each core where a photo shows two cores, as they do?

I've tried to change as little as possible of the original project. I removed a lot of experimental code from the mask generator that was creating synthetic images along with synthetic masks. It looked worth revisiting though. The other thing missing is an expression in code of the metrics used to score core quality - we should check in with Mark about that

### CODE REVIEW

**Comment by mobiuscreek** [comment]
`fastai` has its own implementation of finding the optimal learning rate by computing the gradient of the training loss each time it changes the learning rate. That's what it does with the `lr_find()`, which accepts a `start_lr` and an [`end_lr`](https://fastai1.fast.ai/callbacks.lr_finder.html#LRFinder) and it's the reason why it ends up with a seemingly arbitrary value of 5.25E-05 in that case. You can find the process [here](https://sgugger.github.io/how-do-you-find-a-good-learning-rate.html).

The weight decay (wd) is the default used in the fastai tutorial but since the learning rate is taken care of the wd and the rest of the hyperamaters can be tuned with the use of another library.

**Comment by metazool** [comment]
@mobiuscreek thank you, I saw the use of `lr_find` and wondered why it wasn't returning a value / why an `lr` value was later being set explicitly anyway? I've found this writeup helpful as well: https://platform.ai/blog/page/1/optimizing-hyperparams-for-image-datasets-in-fastai/

So in essence we parameterise all inputs anyway and refine things when we have more annotated data. I'm all for shifting to fastai v2 from this stage onwards. You ok in principle to approve this PR, we draw a line under the original and move to v2 when we package for use with mlflow?

**Comment by mobiuscreek** [comment]
Looks good to me with the only outstanding issue to move it to fastai v2 as soon as the documentation is up to date with the latest code changes.
